Kernel/Process.cpp

```diff
@@ -2367,101 +2367,137 @@ int Process::sys$setsockopt(const Syscall::SC_setsockopt_params* params)
 }
 
 struct SharedBuffer {
-    SharedBuffer(pid_t pid1, pid_t pid2, int size)
-        : m_pid1(pid1)
-        , m_pid2(pid2)
+private:
+    struct Reference {
+        Reference(pid_t pid, unsigned ref, Region* region)
+            : pid(pid)
+            , ref(ref)
+            , region(region)
+        {
+        }
+
+        pid_t pid;
+        unsigned ref;
```

Comment:
Maybe call this `ref_count` or even just `count`? `ref.ref` below looks a bit strange.